src/codec/meta/actor_settings.rs

```diff
@@ -1,22 +1,22 @@
 use ecdsa::signature::rand_core::CryptoRngCore;
 use futures::{AsyncWrite, AsyncWriteExt};
-use winnow::binary::le_u8;
-use winnow::token::take;
-use winnow::Parser;
+use winnow::{binary::le_u8, token::take, Parser};
 
-use crate::codec::crypto::{
-    AccessKey, AsymLockedAccessKey, AsymLockedAccessKeyError, SigningKey, VerifyingKey,
+use crate::codec::{
+    crypto::{AccessKey, AsymLockedAccessKey, AsymLockedAccessKeyError, SigningKey, VerifyingKey},
+    header::AccessMask,
+    meta::{UserAgent, VectorClockActor, VectorClockActorSnapshot},
+    ParserResult, Stream,
 };
-use crate::codec::header::AccessMask;
-use crate::codec::meta::{UserAgent, VectorClock, VectorClockSnapshot};
-use crate::codec::{ParserResult, Stream};
+
+use super::ActorId;
 
 const KEY_PRESENT_BIT: u8 = 0b0000_0001;
 
 #[derive(Clone, Debug)]
 pub struct ActorSettings {
     verifying_key: VerifyingKey,
-    vector_clock: VectorClock,
+    vector_clock: VectorClockActor,
```

Comment:
I suspect we can generally use the snapshot here. Much like the current signing key, we only need the current actor's clock to be consumable during runtime. The only time other Actor's clocks are updated from the perspective of one actor is when we receive journal updates.

Since the clocks are already atomic under the hood, we likely want to keep the "active" actor's clock next to the signing key in the Drive itself... Though we do want it generally accessible whenever an action triggers an operation and those all need the DriveInner handle... We could use a VecDequeue for the specific actor and allow it to get consumed in order assigning clock updates and signatures to the updates as they get processed though...